db/migrate/20240115032403_create_declaration_status_enum.rb

```diff
@@ -0,0 +1,13 @@
+class CreateDeclarationStatusEnum < ActiveRecord::Migration[7.1]
+  def up
+    execute <<-SQL
+      CREATE TYPE declaration_status_enum AS ENUM ('eligible', 'payable', 'paid', 'voided', 'ineligible', 'awaiting_clawback', 'clawed_back');
```

Comment:
Any reason not to use `create_enum(:declaration_status, %w[eligible payable ...])` here?

Reply:
not the biggest fan of including the type of things as part of the name too, it's easy to see within PostgreSQL.

The states etc look fine though